### PR TITLE
Add Documentation for Deployment Targets API

### PR DESCRIPTION
**Pull request by dexyk**

### CODE REVIEW

File: docs/reference/api-v1alpha.md [hunk 1]
```diff
@@ -916,4 +916,540 @@ curl -i \
...
+**Request Body**
+
+The request body should be a JSON object, encapsulating details about the deployment target to be created. The available fields are as follows:
+
+  - `name` (**required**) - Unique name for the target within the project.
+  - `project_id` (**required**) - UUID of the project.
+  - `unique_token` (**required**) - Idempotency UUID token.
+  - `description` (*optional*) - A description of the target.
+  - `url` (*optional*) - The URL of the target.
+  - `bookmark_parameter1`, `bookmark_parameter2`, `bookmark_parameter3` (*optional*) - Bookmark parameters.
```
Comment (DamjanBecirovic) [inline docs/reference/api-v1alpha.md:1150]: Let's expand the description of what bookmarks are here. You can reuse some parts from the CLI docs.

File: docs/reference/api-v1alpha.md [hunk 2]
```diff
@@ -916,4 +916,540 @@ curl -i \
...
+  - `subject_rules` (*optional*) - A list of subject rules structured as `{"type": RULE_TYPE, "subject_id": ID}`. The `RULE_TYPE` can be one of `"ANY"`, `"USER"`, `"ROLE"`, `"GROUP"`, `"AUTO"` and the `ID` should be the name of the role or group (which must be valid for the project) or UUID of a user. For `USER` type you can use `{"type": "USER", "git_login": GITLOGIN}` where `GITLOGIN` is a user's git handle. The user must be assigned to the project to be used in the rule.
```
Comment (DamjanBecirovic) [inline docs/reference/api-v1alpha.md:1151]: Let's remove `"GROUP"` here until it is fully implemented.
On the other hand, we should add a line that describes what  `"AUTO"` is (if auto-promotions are allowed) and how to use it.

File: docs/reference/api-v1alpha.md [hunk 3]
```diff
@@ -916,4 +916,540 @@ curl -i \
...
+  - `object_rules` (*optional*) - A list of object rules structured as `{"type": RULE_TYPE, "match_mode": MODE, "pattern": PATTERN}`. The `RULE_TYPE` can be one of `"BRANCH"`, `"TAG"`, `"PR"` and the `MODE` can be `"ALL"`, `"EXACT"`, `"REGEX"`.
```
Comment (DamjanBecirovic) [inline docs/reference/api-v1alpha.md:1152]: Add a line on what are the limitations for `"PR"` as a type.

File: docs/reference/api-v1alpha.md [hunk 4]
```diff
@@ -916,4 +916,540 @@ curl -i \
...
+```
+GET {org_name}.semaphoreci.com/api/v1alpha/deployment_targets/:target_id/history
+```
+
+**Params**
+
+- `target_id` (**required**) - The UUID of the deployment target for which the deployment history is being retrieved.
+- `cursor_type` (*optional*) - Defines the starting point for data retrieval. Possible values: `"FIRST"`, `"AFTER"`, `"BEFORE"`.
+- `cursor_value` (*optional*) - Specific value for the data retrieval cursor to refer to. Works in conjunction with `cursor_type`.
+- `filters` (*optional*) - Filters that you may apply to narrow down the returned history results.
```
Comment (DamjanBecirovic) [inline docs/reference/api-v1alpha.md:1399]: We should list all available filters here and add short explanation for each of them.

File: docs/reference/api-v1alpha.md [hunk 5]
```diff
@@ -916,4 +916,546 @@ curl -i \
...
+  - `unique_token` (**required**) - Idempotency UUID token.
+  - `description` (*optional*) - A description of the target.
+  - `url` (*optional*) - The URL of the target.
+  - `bookmark_parameter1`, `bookmark_parameter2`, `bookmark_parameter3` (*optional*) - Bookmark parameters - string values that represent the names of the promotion parameters. You can later use values of those parameters to [filter deployments in deployment history](/reference/deployment-targets-yaml-reference/#bookmark_parameter1-bookmark_parameter2-bookmark_parameter3).
+  - `subject_rules` (*optional*) - A list of subject rules structured as `{"type": RULE_TYPE, "subject_id": ID}`. The `RULE_TYPE` can be one of `"ANY"`, `"USER"`, `"ROLE"`, `"AUTO"` and the `ID` should be the name of the role (which must be valid for the project) or UUID of a user. For `USER` type you can use `{"type": "USER", "git_login": GITLOGIN}` where `GITLOGIN` is a user's git handle. The user must be assigned to the project to be used in the rule. The promotion is automatically triggered for the `"AUTO"` rule, provided it is allowed. In the case of `ANY` and `AUTO` rules, there is no need to specify the `subject_id`.
```
Comment (DamjanBecirovic) [inline docs/reference/api-v1alpha.md:1151]: Let's also add a bit of explanation at the begging of the line for what are the subject rules actually. Something like:

> - `subject_rules` (*optional*) - Configures **who** can trigger a promotion of the given deployment target. It should be a  list of elements structured as...

File: docs/reference/api-v1alpha.md [hunk 6]
```diff
@@ -916,4 +916,546 @@ curl -i \
...
+  - `object_rules` (*optional*) - A list of object rules structured as `{"type": RULE_TYPE, "match_mode": MODE, "pattern": PATTERN}`. The `RULE_TYPE` can be one of `"BRANCH"`, `"TAG"`, `"PR"` and the `MODE` can be `"ALL"`, `"EXACT"`, `"REGEX"`. When using the `PR` rule type, the `match_mode` and `pattern` parameters cannot be utilized. Simply including `PR` in the `object_rules` is sufficient to ensure that any pull request will automatically trigger a promotion.
```
Comment (DamjanBecirovic) [inline docs/reference/api-v1alpha.md:1152]: Let's add a bit of explanation at the begging of the line for what are the object rules actually. Something like:

> - `object_rules` (*optional*) - Configures on **which git references** is a promotion of the given deployment target allowed. It should be a  list of elements structured as...

**Review by DamjanBecirovic — APPROVED** [review]

**Review by Charismatron — APPROVED** [review]
Copy edited.

**Pull request merged**